sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCacheNonBlocking.java

```diff
@@ -81,7 +82,7 @@ public Mono<TValue> getAsync(
                     return Mono.just(value);
                 }
 
-                Mono<TValue> refreshMono = initialLazyValue.createAndWaitForBackgroundRefreshTaskAsync(key, singleValueInitFunc);
+                Mono<TValue> refreshMono = initialLazyValue.getOrCreateBackgroundRefreshTaskAsync(singleValueInitFunc);
 
                 return refreshMono.onErrorResume(
                     (exception) -> {
```

Comment:
If for some reason the below logger statement throws an error, we would never remove the value from cache. 
I think just to be safe, we should first remove the values from cache in these `onErrorResume` blocks, and then log the exception or do less important work. 
Same applies for other `onErrorResume` and `doOnError` blocks.

Reply:
make sense, will change in next iterator

Reply:
updated